### PR TITLE
Added list functionality

### PR DESCRIPTION
**Pull request by zskamljic**
Added functionality to inject lists

### CODE REVIEW

File: injector/src/main/java/com/equaleyes/injector/Injector.java [hunk 1]
```diff
@@ -35,31 +40,105 @@ public static void injectToFrom(Object viewsContainer, View view) {
...
+                            Array.set(array, i, downcastItem(views[i]));
+
+                        field.set(owner, array);
+                    }
+                } else if (field.getType() == List.class) {
+                    ParameterizedType type = (ParameterizedType) field.getGenericType();
+                    if (type.getActualTypeArguments()[0] == View.class)
+                        field.set(owner, Arrays.asList(views));
+                    else {
+
```
Comment (dejanskledar) [inline injector/src/main/java/com/equaleyes/injector/Injector.java]: Unneeded empty line

**Comment by dejanskledar** [comment]
👍

File: injector/src/main/java/com/equaleyes/injector/Injector.java [hunk 2]
```diff
@@ -35,31 +40,104 @@ public static void injectToFrom(Object viewsContainer, View view) {
...
-            int id = Util.findDesiredId(toInject.id(), field.getName(), owner);
+            int id = Util.findDesiredId(toInject.value(), field.getName(), owner);
+            if (id == -1) {
+                Log.w("INJECTOR", "Warning: Field \"" + field.getName() + "\" was not injected.");
+                return;
+            }
+
+            try {
+                View view = findAndSet(container, id, onClickListener);
+                if (view == null)
```
Comment (PrimozPernat) [inline injector/src/main/java/com/equaleyes/injector/Injector.java]: Where are the { } even in one liners

**Comment by roknovak** [comment]
👍

File: injectordemo/src/test/java/com/equaleyes/injectordemo/InjectorTestGroup.java [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+        assertEquals(retrievedListener, listener);
+
+        // Make sure the other two are not affected
+        for (int i = 1; i < 3; i++) {
+            listenerInfo = listenerField.get(values[i]);
+            assertNull(listenerInfo);
+        }
+    }
+
```
Comment (PrimozPernat) [inline injectordemo/src/test/java/com/equaleyes/injectordemo/InjectorTestGroup.java]: only one empty line between methods

File: injectordemo/src/test/java/com/equaleyes/injectordemo/InjectorTestGroup.java [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+        assertEquals(retrievedListener, listener);
+
+        // Make sure the other two are not affected
+        for (int i = 1; i < 3; i++) {
+            listenerInfo = listenerField.get(holder.views[i]);
+            assertNull(listenerInfo);
+        }
+    }
+
+
```
Comment (PrimozPernat) [inline injectordemo/src/test/java/com/equaleyes/injectordemo/InjectorTestGroup.java]: empty line

File: injectordemo/src/main/res/layout/item_array_view.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/text1"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:clickable="true"/>
```
Comment (PrimozPernat) [inline injectordemo/src/main/res/layout/item_array_view.xml]: use auto format  android:clickable="true" />

File: injector/src/main/java/com/equaleyes/injector/Injector.java [hunk 6]
```diff
@@ -16,50 +21,133 @@ public static void inject(Activity activity) {
...
+                            Util.getAndroidRId(field.getName()), onClickListener);
+                }
+            }
+
+            try {
+                if (field.getType().isArray()) {
+                    if (field.getType() == View[].class) {
+                        field.set(owner, views);
+                    }
+                    else {
```
Comment (PrimozPernat) [inline injector/src/main/java/com/equaleyes/injector/Injector.java]: it should be } else {. The else should not be in new line.

**Pull request merged**